microproxy/protocol/tls.py

```diff
@@ -91,9 +93,76 @@ def create_src_sslcontext(cert, priv_key, alpn_callback=None,
     ssl_ctx.use_certificate(cert)
     ssl_ctx.use_privatekey(priv_key)
 
-    if alpn_callback:
+    if alpn_callback and HAS_ALPN:
         ssl_ctx.set_alpn_select_callback(alpn_callback)
     if info_callback:
         ssl_ctx.set_info_callback(info_callback)
 
     return ssl_ctx
+
+
+class ServerConnection(object):
+    def __init__(self, stream):
+        self.stream = stream
+        self.alpn_resolver = None
+        self.on_alpn = None
+
+    def start_tls(self, cert, priv_key, info_callback=None,
+                  on_alpn=None, alpn_resolver=None):
+        self.alpn_resolver = alpn_resolver
+        self.on_alpn = on_alpn
+        ssl_ctx = create_src_sslcontext(
```

Comment:
Another question about this part is that why passing two functions instead of just on_alpn? Can you briefly describe the design philosophy behind this part, thanks.

- why passing on_alpn and alpn_resolver
- why not just passing on_alpn as create_src_sslcontext, instead of using self.alpn_callback. 

Thanks